ppa/archive/management/commands/index_pages.py

```diff
@@ -0,0 +1,106 @@
+'''
+Custom multiprocessing Solr index script for page index data.
+'''
+
+import queue
+from multiprocessing import Process, Queue, cpu_count
+from time import sleep
+
+from django.core.management.base import BaseCommand
+from parasolr.django import SolrClient, SolrQuerySet
+import progressbar
+
+from ppa.archive.models import DigitizedWork, Page
+
+
+def page_index_data(work_q, page_data_q):
+    '''Function to generate page index data and add it to
+    a queue. Takes a queue with digitized works to generate pages
+    for and a queue where page data will be added.'''
+    while True:
```

Comment:
```suggestion
    while not work_q.empty():
```
I assume that there was some logic behind why we shouldn't do this? I have an irrational fear of `while True`s

Reply:
This would mean calling the empty check on the queue every iteration, which is extra processing that we don't actually need. Also, apparently empty is not completely reliable "because of multithreading/multiprocessing semantics" ?! 🤨